controllers/secretsync/secret_sync.go

```diff
@@ -0,0 +1,137 @@
+// Copyright (c) 2021 Red Hat, Inc.
+// Copyright Contributors to the Open Cluster Management project
+
+package secretsync
+
+import (
+	"context"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+const (
+	ControllerName = "secret-sync"
+	SecretName     = "policy-encryption-key"
+)
+
+var log = logf.Log.WithName(ControllerName)
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *SecretReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&corev1.Secret{}).
+		Complete(r)
+}
+
+// blank assignment to verify that ReconcilePolicy implements reconcile.Reconciler
+var _ reconcile.Reconciler = &SecretReconciler{}
+
+type SecretReconciler struct {
+	client.Client
+	ManagedClient client.Client
+	Scheme        *runtime.Scheme
+}
+
+// WARNING: In production, this should be namespaced to the actual managed cluster namespace.
+//+kubebuilder:rbac:groups=core,resources=secrets,verbs=create
+//+kubebuilder:rbac:groups=core,resources=secrets,resourceNames=policy-encryption-key,verbs=delete;get;update
+
+// Reconcile handles updates to the "policy-encryption-key" Secret in the managed cluster namespace on the Hub.
+// The method is responsible for synchronizing the Secret to the managed cluster namespace on the managed cluster.
+func (r *SecretReconciler) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
+	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+	reqLogger.V(1).Info("Reconciling Secret")
+	// The cache configuration of SelectorsByObject should prevent this from happening, but add this as a precaution.
+	if request.Name != SecretName {
+		log.Info("Got a reconciliation request for an unexpected Secret. This should have been filtered out.")
```

Comment:
You can filter it with mapper if needed.

Reply:
The cache configuration in `main.go` sets `SelectorsByObject` so only the secret called `policy-encryption-key` is watched. I just added this additional code in case there ends up being a bug in that in the future.

Reply:
Nice!